데이터베이스로 카산드라 사용

lens0021 opened the pull request:
- 기존에 데이터베이스 설정이 기본값(SQLite)이었던 것을 카산드라로 바꾸고 미디어위키 서버와 통신하도록 변경
- secret.php을 볼륨으로 불러오게 하여 카산드라 접속에 필요한 민감 정보들을 사용함
- 도커파일 일부가 길어져 run 스크립트로 분리

#### 관련 이슈

- https://github.com/femiwiki/femiwiki/issues/52

lens0021 committed:
Use cassandra as database

simnalamburt commented on run:
```diff
@@ -0,0 +1,29 @@
+#!/usr/local/bin/node
```
`#!/usr/bin/env node` 로 수정해주세요

simnalamburt reviewed (approved):
그 외엔 LGTM. `#!/usr/bin/env node` 만 고치고 바로 머지해쥬세요

simnalamburt reviewed (approved):
LGTM

simnalamburt commented on run:
```diff
@@ -0,0 +1,29 @@
+const fs = require('fs');
+
+fs.readFile('/a/secret.php', 'utf8', (err, data) => {
+  if (err) throw err;
```
linter를 안붙였기때문에 할말은 없는데, 웬만하면 한줄짜리 if문에 중괄호 생략라지마세요. `if (err) { throw err; }` 이렇게해주세요. 중괄호생략때문에 실수해서 생기는 버그 많아요 https://www.imperialviolet.org/2014/02/22/applebug.html

lens0021 replied:
[https://github.com/femiwiki/restbase/commit/6586260737baa305ef9fc272aaa346a2d8d35ea5](https://github.com/femiwiki/restbase/commit/6586260737baa305ef9fc272aaa346a2d8d35ea5)로 수정했습니다

simnalamburt commented on run:
```diff
@@ -0,0 +1,29 @@
+    data = data.replace(/PARSOID_URI/g, process.env.PARSOID_URI || 'http://parsoid:8000');
+    data = data.replace(/MEDIAWIKI_APIS_DOMAIN/g, process.env.MEDIAWIKI_APIS_DOMAIN || 'femiwiki.com');
+
+    fs.writeFile(configFile, data, (err) => {
```
`(err) => ...` 는 `err => ...`로 바꿔주세요

lens0021 replied:
[https://github.com/femiwiki/restbase/commit/c079acc012b813f49f8c91dbd411967e9f06ebd5](https://github.com/femiwiki/restbase/commit/c079acc012b813f49f8c91dbd411967e9f06ebd5) 로 수정했습니다

simnalamburt commented on run:
```diff
@@ -0,0 +1,29 @@
+
+const fs = require('fs');
+
+fs.readFile('/a/secret.php', 'utf8', (err, data) => {
```
callback 쓰지말고 async/await으로 짭시다 

https://nodejs.org/api/fs.html#fs_fspromises_readfile_path_options

```js
const fs = require('fs').promises;
const data = await fs.readFile('/a/secret.php', 'utf8');
```

simnalamburt reviewed (changes requested):
리뷰 취소. 제가 요청드린거 수정해주세요

lens0021 committed:
Use braces with single-line if blocks

lens0021 committed:
Omit parentheses for arrow function has only one parameter

lens0021 committed:
Use async/await instead of callback

lens0021 committed:
Capitalize some constants

simnalamburt committed:
Update run script

1.  Use strict mode
2.  Misc optimization regarding promises

Reference:
  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode
